File: atcoder.jp/Python (3.8.2)/abc262/abc262_b.py
from collections import defaultdict
N, M = map(int,input().split())

graph = defaultdict(set)
for _ in range(M):
    U, V = map(int,input().split())
    graph[U].add(V)
    graph[V].add(U)

ans = 0
for a in range(1, N+1):
    for b in range(a+1, N+1):
        for c in range(b+1, N+1):
            if b in graph[a] and c in graph[b] and a in graph[c]:
                ans += 1

print(ans)

File: atcoder.jp/Rust (rustc 1.70.0)/abc262/abc262_b.rs
use proconio::{input, fastout, marker::Usize1};

#[fastout]
fn main() {
    input! {
        n: usize,
        m: usize,
    }

    let mut graph = vec![vec![false; n]; n];

    for _ in 0..m {
        input! {
            a: Usize1,
            b: Usize1,
        }
        graph[a][b] = true;
        graph[b][a] = true;
    }

    let mut ans = 0;
    for a in 0..n {
        for b in a+1..n {
            for c in b+1..n {
                if graph[a][b] && graph[b][c] && graph[c][a] {
                    ans += 1;
                }
            }
        }
    }

    println!("{}", ans);
}
